pkg/workload/pgx_helpers.go

```diff
@@ -319,32 +315,57 @@ func (m *MultiConnPool) WarmupConns(ctx context.Context, numConns int) error {
 	// (128).
 	g.SetLimit(100)
 
-	var warmupConnsPerPool []int
+	type warmupPool struct {
+		maxConns int
+		pool     *pgxpool.Pool
+	}
+	warmupPools := make([]warmupPool, len(m.Pools))
 	if numConns == 0 {
-		warmupConnsPerPool = make([]int, len(m.Pools))
 		for i, p := range m.Pools {
-			warmupConnsPerPool[i] = int(p.Config().MaxConns)
+			warmupPools[i].maxConns = int(p.Config().MaxConns)
+			warmupPools[i].pool = p
 		}
 	} else {
-		warmupConnsPerPool = distribute(numConns, len(m.Pools))
+		numPoolConns := distribute(numConns, len(m.Pools))
 		for i, p := range m.Pools {
-			poolMaxConns := int(p.Config().MaxConns)
-			if warmupConnsPerPool[i] > poolMaxConns {
-				warmupConnsPerPool[i] = poolMaxConns
+			warmupPools[i].pool = p
+			warmupPools[i].maxConns = numPoolConns[i]
+			maxPoolConns := int(p.Config().MaxConns)
```

Comment:
This isn't new code, but I'm wondering why we are not passing `MaxConns` into `distribute` but instead are allowing `distribute` to generate numbers that are larger than what we want, which will then result in the total number of conns being lower than `numConns`?